src/libraries/System.Private.CoreLib/src/System/Enum.cs

```diff
@@ -445,9 +436,6 @@ public bool HasFlag(Enum flag)
                     }
 
 #if RARE_ENUMS
-                case CorElementType.ELEMENT_TYPE_BOOLEAN:
```

Comment:
The JIT is going to optimize this case https://github.com/dotnet/runtime/search?q=gtOptimizeEnumHasFlag as intrinsic. This should stay so that we do have mismatch between intrinsic implementation in the JIT and the library implementation here.